packages/next/client/head-manager.ts

```diff
@@ -84,14 +88,25 @@ export default function initHeadManager() {
     updateHead: (head: JSX.Element[]) => {
       const promise = (updatePromise = Promise.resolve().then(() => {
         if (promise !== updatePromise) return
-
         updatePromise = null
         const tags: Record<string, JSX.Element[]> = {}
 
         head.forEach((h) => {
-          const components = tags[h.type] || []
-          components.push(h)
-          tags[h.type] = components
+          // handle react components
+          if (typeof h.type === 'function') {
+            const el = parse(ReactDOMServer.renderToStaticMarkup(h))
```

Comment:
You can just pass the `head` array directly to `ReactDOMServer.renderToStaticMarkup`. We don't need to `parse` it on the client-side either. The reason we parse it on the server is because we need to add the `data-next-head` attribute so that we can mark which nodes should be managed by Next.js after the client rehydrates, but we don't need to do that any more after hydration (since we can maintain references to the nodes we create in this function).

Reply:
> You can just pass the head array directly to ReactDOMServer.renderToStaticMarkup.

<img width="1484" alt="Screen Shot 2020-11-20 at 4 18 08 PM" src="https://user-images.githubusercontent.com/6808172/99855249-fbaea380-2b4b-11eb-9ce3-08f358be49fb.png">

Reply:
or you thinking of something like this?

`const headers = head.map(ReactDOMServer.renderToStaticMarkup)`

Reply:
Sorry for the delay. You should be able to do `ReactDOMServer.renderToStaticMarkup(<React.Fragment>{head}</React.Fragment>)`